api/v1alpha1/accesslogging_types.go

```diff
@@ -71,13 +77,17 @@ const (
 // ProxyAccessLogSink defines the sink of accesslog.
 // +union
 //
+// +kubebuilder:validation:XValidation:rule="self.type == 'ALS' ? has(self.als) : !has(self.als)",message="If AccessLogSink type is ALS, als field needs to be set."
```

Comment:
can you add some test under https://github.com/envoyproxy/gateway/tree/main/test/cel-validation to make sure that this CEL works as expected?

Reply:
added